Can't build HIE

I am on a mac, i deleted the directory an cloned everything from scratch. fails with:

```
LeanderK@scc-wkit-clxa-6-158:~/Documents/Haskell/haskell-ide-engine% make build-all                                                     
git submodule update --init
stack --stack-yaml=stack-8.2.1.yaml install happy
Copying from /Users/LeanderK/.stack/snapshots/x86_64-osx/nightly-2017-11-24/8.2.1/bin/happy to /Users/LeanderK/.local/bin/happy

Copied executables to /Users/LeanderK/.local/bin:
- happy
stack --stack-yaml=stack-8.2.1.yaml install                                \
		&& cp '/Users/LeanderK/.local/bin/hie' '/Users/LeanderK/.local/bin/hie-8.2.1'    \
		&& cp '/Users/LeanderK/.local/bin/hie-8.2.1' '/Users/LeanderK/.local/bin/hie-8.2'
butcher-1.3.1.1: configure
butcher-1.3.1.1: build
cabal-plan-0.3.0.0: configure
cabal-plan-0.3.0.0: build
czipwith-1.0.1.0: configure
czipwith-1.0.1.0: build
constrained-dynamic-0.1.0.0: configure
constrained-dynamic-0.1.0.0: build
czipwith-1.0.1.0: copy/register
ghc-exactprint-0.5.6.1: configure
constrained-dynamic-0.1.0.0: copy/register
ghc-exactprint-0.5.6.1: build
haddock-library-1.4.4: configure
haddock-library-1.4.4: build
butcher-1.3.1.1: copy/register
haskell-lsp-types-0.4.0.0: configure (lib)
haskell-lsp-types-0.4.0.0: build (lib)
cabal-plan-0.3.0.0: copy/register
cabal-helper-0.8.0.4: configure (lib + exe)
cabal-helper-0.8.0.4: build (lib + exe)
haddock-library-1.4.4: copy/register
haddock-api-2.18.1: configure
haddock-api-2.18.1: build
cabal-helper-0.8.0.4: copy/register
ghc-mod-core-5.9.0.0: configure (lib)
ghc-mod-core-5.9.0.0: build (lib)
ghc-exactprint-0.5.6.1: copy/register
apply-refact-0.4.1.0: configure
apply-refact-0.4.1.0: build
haddock-api-2.18.1: copy/register
hsimport-0.8.6: configure
hsimport-0.8.6: build
hsimport-0.8.6: copy/register
sorted-list-0.2.1.0: configure
sorted-list-0.2.1.0: build
sorted-list-0.2.1.0: copy/register
syz-0.2.0.0: configure
apply-refact-0.4.1.0: copy/register
syz-0.2.0.0: build
yaml-0.8.32: configure
syz-0.2.0.0: copy/register
yaml-0.8.32: build
ghc-mod-core-5.9.0.0: copy/register
HaRe-0.8.4.1: configure (lib + exe)
HaRe-0.8.4.1: build (lib + exe)
haskell-lsp-types-0.4.0.0: copy/register
haskell-lsp-0.4.0.0: configure (lib + exe)
haskell-lsp-0.4.0.0: build (lib + exe)
yaml-0.8.32: copy/register
hlint-2.0.11: configure
hlint-2.0.11: build
brittany-0.11.0.0: configure
brittany-0.11.0.0: build
HaRe-0.8.4.1: copy/register
haskell-lsp-0.4.0.0: copy/register
hie-plugin-api-0.2.0.0: configure (lib)
hie-plugin-api-0.2.0.0: build (lib)
hlint-2.0.11: copy/register
ghc-mod-5.9.0.0: configure (lib + exe)
ghc-mod-5.9.0.0: build (lib + exe)
hie-plugin-api-0.2.0.0: copy/register
ghc-mod-5.9.0.0: copy/register
brittany-0.11.0.0: copy/register
Building all executables for `haskell-ide-engine' once. After a successful build of all of them, only specified executables will be rebuilt.
haskell-ide-engine-0.2.1.0: configure (lib + exe)
haskell-ide-engine-0.2.1.0: build (lib + exe)
haskell-ide-engine-0.2.1.0: copy/register
Completed 22 action(s).
Log files have been written to: /Users/LeanderK/Documents/Haskell/haskell-ide-engine/.stack-work/logs/
/Users/LeanderK/.stack/setup-exe-cache/x86_64-osx/Cabal-simple_mPHDZzAJ_2.0.0.2_ghc-8.2.1: streamingProcess: invalid argument (Bad file descriptor)
make: *** [hie-8.2.1] Error 1
```

--- a/.circleci/config.yml
+++ b/.circleci/config.yml
@@ -39,8 +39,8 @@ defaults: &defaults
         command: stack --stack-yaml=${STACK_FILE} install hoogle
 
     - run:
-        name: Compile but do not run tests
-        command: stack -j 2 --stack-yaml=${STACK_FILE} test --no-run-tests
+        name: Build
+        command: stack -j 2 --stack-yaml=${STACK_FILE} build
 
     - run:
         name: Generate Hoogle database
@@ -56,7 +56,7 @@ defaults: &defaults
           key: stack-work-dirs-04-{{ arch }}-{{ .Environment.CIRCLE_JOB }}-{{ checksum "stack-build.txt" }}
 
     - run:
-        name: Compile + Test
+        name: Test
         command: stack -j 1 --stack-yaml=${STACK_FILE} test --dump-logs
         no_output_timeout: 120m
 

--- a/haskell-ide-engine.cabal
+++ b/haskell-ide-engine.cabal
@@ -139,6 +139,7 @@ executable hie-wrapper
 test-suite unit-test
   type:                exitcode-stdio-1.0
   hs-source-dirs:      test/unit
+                     , test/utils
   main-is:             Main.hs
   other-modules:       ApplyRefactPluginSpec
                        BrittanySpec
@@ -150,6 +151,8 @@ test-suite unit-test
                        HooglePluginSpec
                        JsonSpec
                        Spec
+
+                       TestUtils
   build-depends:       QuickCheck
                      , aeson
                      , base
@@ -158,13 +161,21 @@ test-suite unit-test
                      , filepath
                      , haskell-lsp
                      , haskell-ide-engine
-                     , hie-test-utils
+                     -- , hie-test-utils
                      , hie-plugin-api
                      , hoogle > 5.0.11
                      , hspec
                      , quickcheck-instances
                      , text
                      , unordered-containers
+
+                     -- remove these once hie-test-utils is reinstated
+                     , data-default
+                     , hie-plugin-api
+                     , ghc-mod-core
+                     , hslogger
+                     , yaml
+
   ghc-options:         -threaded -rtsopts -with-rtsopts=-N -Wall -Wredundant-constraints
   if flag(pedantic)
      ghc-options:      -Werror
@@ -174,7 +185,9 @@ test-suite unit-test
 test-suite dispatcher-test
   type:                exitcode-stdio-1.0
   hs-source-dirs:      test/dispatcher
+                       test/utils
   main-is:             Main.hs
+  other-modules:       TestUtils
   build-depends:       base
                      , aeson
                      , containers
@@ -183,44 +196,52 @@ test-suite dispatcher-test
                      , filepath
                      , haskell-lsp
                      , haskell-ide-engine
-                     , hie-test-utils
+                     -- , hie-test-utils
                      , hie-plugin-api
                      , hspec
                      , stm
                      , text
                      , unordered-containers
+
+                     -- remove these once hie-test-utils is reinstated
+                     , hie-plugin-api
+                     , ghc-mod-core
+                     , hslogger
+                     , unordered-containers
+                     , yaml
   ghc-options:         -threaded -rtsopts -with-rtsopts=-N -Wall -Wredundant-constraints
   if flag(pedantic)
      ghc-options:      -Werror
   default-language:    Haskell2010
   build-tool-depends:  hspec-discover:hspec-discover
 
-library hie-test-utils
-  hs-source-dirs:      test/utils
-  exposed-modules:     TestUtils
-  build-depends:       base
-                     , haskell-ide-engine
-                     , haskell-lsp
-                     , hie-plugin-api
-                     , aeson
-                     , containers
-                     , data-default
-                     , directory
-                     , filepath
-                     , ghc-mod-core
-                     , hslogger
-                     , hspec
-                     , text
-                     , unordered-containers
-                     , yaml
-  ghc-options:         -Wall -Wredundant-constraints
-  if flag(pedantic)
-     ghc-options:      -Werror
-  default-language:    Haskell2010
+-- library hie-test-utils
+--   hs-source-dirs:      test/utils
+--   exposed-modules:     TestUtils
+--   build-depends:       base
+--                      , haskell-ide-engine
+--                      , haskell-lsp
+--                      , hie-plugin-api
+--                      , aeson
+--                      , containers
+--                      , data-default
+--                      , directory
+--                      , filepath
+--                      , ghc-mod-core
+--                      , hslogger
+--                      , hspec
+--                      , text
+--                      , unordered-containers
+--                      , yaml
+--   ghc-options:         -Wall -Wredundant-constraints
+--   if flag(pedantic)
+--      ghc-options:      -Werror
+--   default-language:    Haskell2010
 
 test-suite func-test
   type:                exitcode-stdio-1.0
   hs-source-dirs:      test/functional
+                     , test/utils
   main-is:             Main.hs
   other-modules:       CompletionSpec
                      , CodeActionsSpec
@@ -228,8 +249,9 @@ test-suite func-test
                      , ReferencesSpec
                      , RenameSpec
                      , Spec
-  build-depends:       hie-test-utils
-                     , aeson
+
+                     , TestUtils
+  build-depends:       aeson
                      , base
                      , containers
                      , data-default
@@ -237,12 +259,21 @@ test-suite func-test
                      , filepath
                      , haskell-lsp-test
                      , haskell-ide-engine
+                     -- , hie-test-utils
                      , hie-plugin-api
                      , hspec
                      , lens
                      , stm
                      , text
                      , unordered-containers
+
+                     -- remove these once hie-test-utils is reinstated
+                     , hie-plugin-api
+                     , ghc-mod-core
+                     , hslogger
+                     , unordered-containers
+                     , yaml
+                     , haskell-lsp
   ghc-options:         -threaded -rtsopts -with-rtsopts=-N -Wall -Wredundant-constraints
   if flag(pedantic)
      ghc-options:      -Werror

